cli/cmd/install.go

```diff
@@ -399,18 +399,23 @@ func renderCRDs(ctx context.Context, k *k8s.KubernetesAPI, w io.Writer, options
 		return err
 	}
 
+	finalValues := charts.MergeMaps(defaultValues, valuesOverrides)
+
 	// If any of the Gateway API CRDs are installed, we default to rendering the
 	// Gateway API CRDs.
 	if k != nil {
 		installed, err := checkGatewayAPICRDs(ctx, k)
 		if err != nil {
 			return err
 		}
-		if installed == Absent {
+		if installed == Absent && finalValues["installGatewayAPI"] == false {
```

Comment:
something to be cautious of here is that we compute finalValues based on defaultValue, but then later can update defaultValues without that change getting reflected in finalValues.  It doesn't specifically impact the logic here because defaultValues is only updated in if-branches where we don't use finalValues but it will make it very easy for bugs to creep in.

To do this right, I think defaulting and validation need to be done in discrete steps instead of trying to do them all at once as I tried in the original change. 

i.e

1. Update defaultValues based on the value of `installed`
2. Compute finalValues based on defaultValues and valuesOverrides
3. Validate finalValues against the value of `installed